de.dashup.application/src/main/java/de/dashup/application/controllers/WorkbenchController.java

```diff
@@ -0,0 +1,278 @@
+            return "redirect:/workbench/draft/" + draft.getId();
+        }
+        return "redirect:/login";
+    }
+
+    @RequestMapping("/draft/{draftId}")
+    public String workbenchDraft(@CookieValue(name = "token", required = false) String token,
+                                 HttpServletRequest request, Model model,
+                                 @PathVariable(value = "draftId") int draftId) throws SQLException {
+        return ControllerHelper.defaultMapping(token, request, model, "workbench", user -> {
+            DashupService.getInstance().getUsersDrafts(user);
+            model.addAttribute("drafts", user.getDrafts());
+            model.addAttribute("publishedWidgets", DashupService.getInstance().getUsersWidgets(user));
+            model.addAttribute("categories", Widget.Category.values());
+            model.addAttribute("tags", DashupService.getInstance().getAllTags());
+            DashupService.getInstance().getSectionsAndPanels(user);
+            model.addAttribute("sections", user.getSections());
+
+            for (Draft draft : user.getDrafts()) {
+                if (draft.getId() == draftId) {
+                    model.addAttribute("current", draft);
+                    DashupService.getInstance().getTagsByWidget(draft);
+                    model.addAttribute("currentTags", draft.getTags());
+                    break;
+                }
+            }
+        });
+    }
+
+    @RequestMapping("/published/{publishedWidgetId}")
+    public String workbenchPublished(@CookieValue(name = "token", required = false) String token,
+                                     HttpServletRequest request, Model model,
+                                     @PathVariable(value = "publishedWidgetId") int publishedWidgetId) throws SQLException {
+        return ControllerHelper.defaultMapping(token, request, model, "workbench", user -> {
+            DashupService.getInstance().getUsersDrafts(user);
+            model.addAttribute("drafts", user.getDrafts());
+            List<Widget> publishedWidgets = DashupService.getInstance().getUsersWidgets(user);
+            model.addAttribute("publishedWidgets", publishedWidgets);
+            model.addAttribute("categories", Widget.Category.values());
+            model.addAttribute("tags", DashupService.getInstance().getAllTags());
+            DashupService.getInstance().getSectionsAndPanels(user);
+            model.addAttribute("sections", user.getSections());
+
+
+            for (Widget widget : publishedWidgets) {
+                if (widget.getId() == publishedWidgetId) {
+                    model.addAttribute("current", widget);
+                    DashupService.getInstance().getTagsByWidget(widget);
+                    model.addAttribute("currentTags", widget.getTags());
+                    break;
+                }
+            }
+        });
+    }
+
+    @RequestMapping(value = "/draft/{draftId}/deleteDraft")
```

Comment:
We should maybe make sure, that the user making the delete request is the user, who created the panel. We do not want to have Jannik delete all our widgets mid presentation.